hazelcast-sql-core/src/test/java/com/hazelcast/sql/impl/expression/math/MinusOperatorIntegrationTest.java

```diff
@@ -387,30 +399,175 @@ public void testDouble() {
     }
 
     @Test
-    public void testTemporal() {
-        checkUnsupportedForAllTypesCommute(LOCAL_DATE_VAL, DATE);
-        checkUnsupportedForAllTypesCommute(LOCAL_TIME_VAL, TIME);
-        checkUnsupportedForAllTypesCommute(LOCAL_DATE_TIME_VAL, TIMESTAMP);
-        checkUnsupportedForAllTypesCommute(OFFSET_DATE_TIME_VAL, TIMESTAMP_WITH_TIME_ZONE);
+    public void testDate() {
+        LocalDate date = LocalDate.now();
+
+        // Check null values when one side is interval
+        putAndCheckValue(date, sql(null, "INTERVAL '1' SECOND"), TIMESTAMP, null);
+        putAndCheckFailure(date, sql("INTERVAL '1' SECOND", null), PARSING, signatureError("INTERVAL_SECOND", TIMESTAMP));
+
+        // Check null values when one side is temporal type. Since we cannot deduce the type of the other hand, we fail.
+        putAndCheckFailure(date, sql("this", null), PARSING, signatureError(DATE, DATE));
+        putAndCheckFailure(date, sql(null, "this"), PARSING, signatureError(DATE, DATE));
+
+        // Check normal operations
+        putAndCheckValue(date, sql("this", "INTERVAL '1' SECOND"), TIMESTAMP, date.atStartOfDay().minus(1, SECONDS));
```

Comment:
I tried to add these tests:
```java
        putAndCheckValue(date, sql("this", "INTERVAL '0.000001' SECOND"), TIMESTAMP, date.atStartOfDay().plus(1, MICROS));
        putAndCheckValue(date, sql("this", "INTERVAL '0.000001' SECOND(6)"), TIMESTAMP, date.atStartOfDay().plus(1, MICROS));
        putAndCheckValue(date, sql("this", "INTERVAL '0.000000001' SECOND(9)"), TIMESTAMP, date.atStartOfDay().plus(1, NANOS));
```
The first two fail with:
```
Expected :2021-03-15T00:00:00.000001
Actual   :2021-03-15T00:00
```
because we support only millisecond resolution.

The last one fails with:
```
org.apache.calcite.runtime.CalciteContextException: From line 1, column 15 to line 1, column 46: Illegal interval literal format '0.000000001' for INTERVAL SECOND(9)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at org.apache.calcite.runtime.Resources$ExInstWithCause.ex(Resources.java:457)
	at org.apache.calcite.sql.SqlUtil.newContextException(SqlUtil.java:857)
	at org.apache.calcite.sql.SqlUtil.newContextException(SqlUtil.java:842)
	at org.apache.calcite.sql.SqlIntervalQualifier.invalidValueException(SqlIntervalQualifier.java:1165)
	at org.apache.calcite.sql.SqlIntervalQualifier.evaluateIntervalLiteralAsSecond(SqlIntervalQualifier.java:1077)
	at org.apache.calcite.sql.SqlIntervalQualifier.evaluateIntervalLiteral(SqlIntervalQualifier.java:1152)
	at org.apache.calcite.sql.validate.SqlValidatorImpl.validateLiteral(SqlValidatorImpl.java:3155)
```
In this case `SECOND(9)` is accepted, but the actual `SqlIntervalQualifier` doesn't have the 9 for precision.

Reply:
Calcite supports only millisecond resolution. We may consider changing/relaxing it, but the proper implementation of this would require additional extensions to the validator and possibly parser. I propose to fix this separately.

Reply:
Isn't it possible to limit the interval literal to 3 decimal places? I guess it's calcite bug then, i'd be fine if we file it there.

Reply:
I filed a bug: https://issues.apache.org/jira/browse/CALCITE-4543